src/dispatchers/MemoryUsageDispatcher.js

```diff
@@ -0,0 +1,65 @@
+/* @flow */
+import type { Dispatcher } from './types';
+import CommandExecutor from '../CommandExecutor';
+import { logDebug } from '../common';
+import {
+  memoryUsageSupported,
+  MEASUREMENT_INTERVAL_IN_MS,
+} from '../memory/utils';
+
+class MemoryUsageDispatcher implements Dispatcher {
+  executors: Array<CommandExecutor>;
+  started: boolean;
+
+  constructor() {
+    this.executors = [];
+    this.started = false;
+  }
+
+  addExecutor(executor: CommandExecutor) {
+    this.executors.push(executor);
+  }
+
+  start() {
+    if (!memoryUsageSupported()) {
+      logDebug('Memory Usage API is not supported, skipping MemoryUsageDispatcher');
+      return;
+    }
+
+    if (!this.started) {
+      this.started = true;
+      // start the first measurement almost right away
+      this.scheduleMeasurement(100);
```

Comment:
I'm thinking out loud here, maybe we should wait until the page is considered loaded (document.readyState = 'complete') to get useful reading of memory usage. IMO it would be more reflective of real page memory usage as all resources will be loaded and parsed by then.

Reply:
Hmmm... that is not a bad idea. Let me modify the logic here.

Reply:
@amir-hadzic I adjusted the code, check now :)

Reply:
Looks good